### PR TITLE
feat(api): Expose Markdown Editor API in Angular Component

### PR DESCRIPTION
**Pull request by jeremysmartt**
## Description
Can't use styles attribute on AOT, need to inject the css

### Test Steps
- [ ] Checkout branch
- [ ] npm install
- [ ] npm run build
- [ ] cd deploy/platform/text-editor
- [ ] npm pack (prints out a tgz file created)
- [ ] pwd (prints out your < path > you are at)
- [ ] Go to Covalent repo
- [ ] rm -rf node_modules/`@covalent`/text-editor
- [ ] npm install < path > / < tgz file created >
- [ ] ng serve --aot
- [ ] In browser go to: http://localhost:4200/#/components/text-editor
- [ ] See Markdown Editor running
- [ ] Use some of the functionality and see works

#### Screenshots
screen shot 2017-08-04 at 2 08 53 pm

### CODE REVIEW

File: docs/SETUP.md [hunk 1]
```diff
@@ -6,7 +6,18 @@ npm install @covalent/text-editor
...
 * Add the following tag to html file
 ```html
-<td-text-editor></td-text-editor>
+<td-text-editor [value]="Some Text" options="opts"></td-text-editor>
+```
+* Fill in options with any options listed here: 
+  * <a href="https://github.com/sparksuite/simplemde-markdown-editor#configuration">https://github.com/sparksuite/simplemde-markdown-editor#configuration</a>
+```typescript
+class MyComponent {
+  opts: any = {
```
Comment (kyleledbetter) [inline docs/SETUP.md]: would it hurt to just call `opts` `options` instead? :)

Comment (jeremysmartt) [reply docs/SETUP.md]: :) sure

File: docs/API.md [hunk 2]
```diff
@@ -4,29 +4,34 @@
...
+| `toTextArea` | `function()` | Reverts to the Initial textarea. Returns void
 
 ### Usage
 
 Example for HTML usage:
 
 ```html
-<td-text-editor>
-</td-text-editor>
+<td-text-editor [value]="Some Text" options="options"></td-text-editor>
```
Comment (emoralesb05) [inline docs/API.md]: `options` should be `[options]` here

File: docs/SETUP.md [hunk 3]
```diff
@@ -6,7 +6,18 @@ npm install @covalent/text-editor
 ```
 * Add the following tag to html file
 ```html
-<td-text-editor></td-text-editor>
+<td-text-editor [value]="Some Text" options="options"></td-text-editor>
```
Comment (emoralesb05) [inline docs/SETUP.md]: same here

File: src/platform/text-editor/text-editor.component.ts [hunk 4]
```diff
@@ -74,11 +75,34 @@ export class TdTextEditorComponent implements AfterViewInit, ControlValueAccesso
   }
 
   ngAfterViewInit(): void {
-    this._simpleMDE = new SimpleMDE({ element: this.elementRef.nativeElement.value });
+    this.options.element = this.elementRef.nativeElement.value;
+    this._simpleMDE = new SimpleMDE(this.options);
```
Comment (emoralesb05) [inline src/platform/text-editor/text-editor.component.ts:79]: does passing an empty object override everything?

**Review by emoralesb05 — CHANGES_REQUESTED** [review]

File: src/platform/text-editor/text-editor.component.ts (hunk 4 above)
Comment (jeremysmartt) [reply src/platform/text-editor/text-editor.component.ts:79]: nope, all good

**Review by emoralesb05 — APPROVED** [review]

**Pull request merged**